src/main/java/org/jabref/gui/openoffice/OpenOfficePanel.java

```diff
@@ -601,12 +587,12 @@ private boolean checkThatEntriesHaveKeys(List<BibEntry> entries) {
         Optional<BibDatabaseContext> databaseContext = stateManager.getActiveDatabase();
         if (citePressed && databaseContext.isPresent()) {
             // Generate keys
-            CitationKeyPatternPreferences prefs = preferences.getCitationKeyPatternPreferences();
+            CitationKeyPatternPreferences citationKeyPatternPreferences = preferences.getCitationKeyPatternPreferences();
```

Comment:
Maybe, you can have these passed in the constructor - and get rid of `preferences` parameter alltogheter? ^^

Reply:
That would actually be the ideal level of abstraction, but I am a little confused on how to do that as perhaps this constructor is called internally (IntelliJ shows no usages):

![image](https://github.com/user-attachments/assets/cf039694-6d07-4ee7-a4f0-669db6f95193)

Reason for moving `openOfficePreferences` in the constructor initialization block was that it is the core preference used all over the class (22 usages), whereas the rest of the preferences (ExternalApplicationsPreferences, LayoutFormatterPreferences, CitationKeyPatternPreferences) are used just used once in within their respective scopes, thus declared locally.

Reply:
Update: Found it. `SidePaneContentFactory`. I'll give this a try.

Reply:
Update: Nope, `preferences` would be needed here:
![image](https://github.com/user-attachments/assets/ce7d93df-c4e1-4358-bd6c-e22d6e43e2f8)

Reply:
@koppor I tried to clean it up a bit further by passing all the preference subtypes via constructor. Although I couldn't get rid of `preferences`, I could minimize its usage to this one place (above comment). Check commit https://github.com/JabRef/jabref/pull/11779/commits/f759087e6880fc90e0c23c632e7dc49a4740f0c8. If this is not useful, we can revert the commit.